src/storage/engine/mod.rs

```diff
@@ -319,6 +319,7 @@ pub struct Cursor<I: Iterator> {
     // the data cursor can be seen will be
     min_key: Option<Vec<u8>>,
     max_key: Option<Vec<u8>>,
+    valid: bool, // whether or not the cursor is hitting the `min_key` and `max_key` bound
```

Comment:
The comments should be "When hit the `min_key` or `max_bound`, should set the valid to false".